### ISSUE TITLE
mtproto.TL_badMsgNotification{BadMsgID:7392862090884919560, BadMsgSeqno:1, ErrorCode:32

### ISSUE THREAD

**Issue by wind-hx** [3bl3gamer/tgclient#28]
I upgraded to the latest code. This error occurs when using the sendMessage method after layer 183.

https://core.telegram.org/mtproto/service_messages_about_messages

msg_seqno too low (the server has already received a message with a lower msg_id but with either a higher or an equal and odd seqno)

**Comment by wind-hx** [comment]
Judging from the error log printed on the console, it is an error that occurred during connect.

**Comment by 3bl3gamer** [comment]
> This error occurs when using the sendMessage

> it is an error that occurred during connect

Does it happen both while sending message and while connecting?

Does it happen during the first connection or during a reconnection after some network error?

Is there a way to reliably reproduce this error? I've tried to reconnect and send message multiple times, it worked OK for me.

**Comment by 3bl3gamer** [comment]
> This error occurs when using the sendMessage

> it is an error that occurred during connect

Does it happen both while sending message and while connecting?

Does it happen during the first connection or during a reconnection (as in https://github.com/3bl3gamer/tgclient/issues/27)? Is this reconnection caused by some other error?

Is there a way to reliably reproduce this error? I've tried to reconnect and send message multiple times, it worked OK for me.

**Comment by wind-hx** [comment]
When the AUTH_KEY_DUPLICATED error occurs during initConnect, there is no need to try to reconnect, because the "mtproto.TL_badMsgNotification{BadMsgID:7392862090884919560, BadMsgSeqno:1, ErrorCode:32" error will appear when reconnecting.

**Comment by wind-hx** [comment]
https://core.telegram.org/api/errors

If the client receives an AUTH_KEY_DUPLICATED error, the session was already invalidated by the server and the user must generate a new auth key and login again.

Program errors when client receives AUTH_KEY_DUPLICATED error

**Comment by 3bl3gamer** [comment]
> When the AUTH_KEY_DUPLICATED error occurs during initConnect, there is no need to try to reconnect

I agree. But does `TL_badMsgNotification{ErrorCode:32}` only occur after `AUTH_KEY_DUPLICATED`? If so, there seems no need to check `ErrorCode:32/33` and try to recover (as in https://github.com/3bl3gamer/tgclient/pull/29)? Since it will not be possible to recover (without re-auth) anyway. And during normal operation there should be no errors with `ErrorCode:32/33`.

**Comment by wind-hx** [comment]
"ErrorCode:32/33" This error does not need to be handled, but I referred to "https://github.com/gram-js/gramjs/blob/9233537e172ffce6d44a09454b7c75369a846bbf/gramjs/network/MTProtoSender.ts#L856" This part of the code

**Comment by wind-hx** [comment]
There is no need to check "ErrorCode:32/33", I just referred to the "GramJS" project

**Comment by wind-hx** [comment]
I found that when an error occurs when the network is unstable, reconnection will be triggered, and "ErrorCode:32" errors will appear infinitely.

```
2024/07/22 18:50:12 [ERROR] reading failed:
read tcp 192.168.224.9:46480->149.154.175.56:443: read: connection reset by peer

wind.com/telegram-transport/src/tgclient/mtproto.(*MTProto).read
	/Users/feit/Project/go-project/telegram-transport-go/src/tgclient/mtproto/network.go:108
wind.com/telegram-transport/src/tgclient/mtproto.(*MTProto).readRoutine
	/Users/feit/Project/go-project/telegram-transport-go/src/tgclient/mtproto/mtproto.go:902
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1695
2024/07/22 18:50:12 [INFO] reconnecting...
2024/07/22 18:50:12 [INFO] reconnecting: DC 1 -> 0
2024/07/22 18:50:12 [INFO] connecting to DC 1 (149.154.175.56:443)...
2024/07/22 18:50:13 [ERROR] msg_seqno too low:
unexpected response: mtproto.TL_badMsgNotification{BadMsgID:7394410744236962940, BadMsgSeqno:1, ErrorCode:32}

wind.com/telegram-transport/src/tgclient/mtproto.(*MTProto).initConection
	/Users/feit/Project/go-project/telegram-transport-go/src/tgclient/mtproto/mtproto.go:288
wind.com/telegram-transport/src/tgclient/mtproto.(*MTProto).Connect
	/Users/feit/Project/go-project/telegram-transport-go/src/tgclient/mtproto/mtproto.go:301
wind.com/telegram-transport/src/tgclient/mtproto.(*MTProto).reconnect
	/Users/feit/Project/go-project/telegram-transport-go/src/tgclient/mtproto/mtproto.go:444
wind.com/telegram-transport/src/tgclient/mtproto.(*MTProto).reconnectLogged
	/Users/feit/Project/go-project/telegram-transport-go/src/tgclient/mtproto/mtproto.go:402
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1695
```
